src/apps/Altinn.AccessManagement/src/Altinn.AccessMgmt.Persistence/Services/AssignmentService.cs

```diff
@@ -29,6 +30,136 @@ IDelegationRepository delegationRepository
     private readonly IRolePackageRepository rolePackageRepository = rolePackageRepository;
     private readonly IEntityRepository entityRepository = entityRepository;
     private readonly IDelegationRepository delegationRepository = delegationRepository;
+    private static readonly string RETTIGHETSHAVER = "rettighetshaver";
+    private static readonly Guid PartyTypeOrganizationUuid = new Guid("8c216e2f-afdd-4234-9ba2-691c727bb33d");
+
+    /// <inheritdoc/>
+    public async Task<IEnumerable<ClientDto>> GetClients(Guid toId, string[] roles, string[] packages, CancellationToken cancellationToken = default)
+    {
+        // Setup filter to be used for fetching client assignments
+        var clientFilter = assignmentRepository.CreateFilterBuilder();
+        clientFilter.Equal(t => t.ToId, toId);
+
+        // Fetch role metadata
+        var roleFilter = roleRepository.CreateFilterBuilder();
+        roleFilter.In(t => t.Code, roles);
+        var roleResult = await roleRepository.Get(roleFilter, cancellationToken: cancellationToken);
+
+        if (roleResult == null || !roleResult.Any() || roleResult.Count() != roles.Length)
+        {
+            throw new ArgumentException($"Filter: {nameof(roles)}, provided contains one or more role identifiers which cannot be found.");
+        }
+
+        var filterRoleIds = roleResult.Select(r => r.Id).ToList();
+
+        // Fetch role-package metadata
+        var rolePackFilter = rolePackageRepository.CreateFilterBuilder();
+        rolePackFilter.In(t => t.RoleId, filterRoleIds);
+        var rolePackageResult = await rolePackageRepository.Get(rolePackFilter, cancellationToken: cancellationToken);
+
+        // Fetch package metadata
+        var packageResult = await packageRepository.Get(cancellationToken: cancellationToken);
```

Comment:
Kan man ikke bruke In filter her for Urn? For her henter vi vel ut alle packages?

Reply:
Gitt at man har oppgitt ett pakkefilter så kunne jeg bruk in-filter på det (hadde også det i starten), men da mister man også kontekst om alle andre pakker man evt. har i tillegg til det man filtrerer. Kanskje ikke egentlig ett problem..

Har man _ikke_ spesifisert pakkefilter så er det i så fall alle pakker som finnes i rolePackage result jeg evt. kan legge i ett In filter.

Reply:
Nei, vent å filtere på rolePackages fungerer ikke. Jeg må evt. også vite alle assignmentPackages

Reply:
Er pga. denne kompleksiteten av jeg endte opp med å fjerne in-filteret her :P